ncm-network/src/main/pan/components/network/types/network/backend/nmstate.pan

```diff
@@ -1,3 +1,60 @@
 declaration template components/network/types/network/backend/nmstate;
 
 @{implement types specific for nmstate / nmstate.pm}
+
+include 'components/network/types/network/route';
+
+type structure_network_nmstate = {
+    @{let NetworkManager manage the dns}
+    "manage_dns" : boolean = false
+    @{let ncm-network cleanup inactive connections}
+    "clean_inactive_conn" : boolean = true
+};
+
+type structure_network_backend_specific = {
+    "nmstate" : structure_network_nmstate
```

Comment:
why the nmstate subtructure? the options have a generic enough name

Reply:
I thought it best to make it clear, but happy to do whichever people prefer.

Reply:
Could we ever have manage_dns and nmstate/manage_dns? If they might conflict then putting it in a subtree makes sense, but creating a nmstate subtree might also be problematic if RH rename nmstate or move the manage_dns out of it?

OTOH, this currently looks to be a single item tree? In which case I am not sure it makes sense to push things down a level? What things do we expect to appear in structure_network_backend_specific versus structure_network_backend_specific/nmstate ?

Reply:
manage_dns was only added during nmstate configuration since network manager by default updates resolve.conf.
 
However upon looking at this few weeks ago for [device] configuration drop in file,
 I think we need schema to configure options for [main] droppin file. one of its option is dns=none, which manage_dns adds in [main] config section, so I wonder if this should be refactored to add any override options of [main] configuration of networkmanager.conf drop in file.

Reply:
Ok cool, I'll drop it.